lms/djangoapps/courseware/user_state_client.py

```diff
@@ -0,0 +1,226 @@
+import itertools
+from operator import attrgetter
+
+try:
+    import simplejson as json
+except:
+    import json
+
+from django.contrib.auth.models import User
+
+from xblock.fields import Scope, ScopeBase
+from xblock_user_state.interface import XBlockUserStateClient
+from courseware.models import StudentModule
+from contracts import contract, new_contract
+from opaque_keys.edx.keys import UsageKey
+
+new_contract('UsageKey', UsageKey)
+
+class DjangoXBlockUserStateClient(XBlockUserStateClient):
+    """
+    An interface that uses the Django ORM StudentModule as a backend.
+    """
+
+    class ServiceUnavailableError(Exception):
+        pass
+
+    class PermissionDeniedError(Exception):
+        pass
+
+    def __init__(self, user):
+        self.user = user
+
+    @contract(username="basestring", block_key=UsageKey, scope=ScopeBase, fields="seq(basestring)|set(basestring)|None")
+    def get(self, username, block_key, scope=Scope.user_state, fields=None):
+        """
+        Retrieve the stored XBlock state for a single xblock usage.
+
+        Arguments:
+            username: The name of the user whose state should be retrieved
+            block_key (UsageKey): The UsageKey identifying which xblock state to load.
+            scope (Scope): The scope to load data from
+            fields: A list of field values to retrieve. If None, retrieve all stored fields.
+
+        Returns
+            dict: A dictionary mapping field names to values
+        """
+        assert self.user.username == username
+        return next(self.get_many(username, [block_key], scope, fields=fields))[1]
```

Comment:
1. Could you do the explicit assignment here, e.g. `_usage_key, state = next(self.get_many(username, [block_key], scope, fields=fields)`
2. What happens if there is no row for `block_key`?